src/context.rs

```diff
@@ -227,12 +260,22 @@ impl<'a> EventCx<'a> {
                 {
                     return true;
                 }
+                // Needed to handle mouse leave for hovered views
+                if self.app_state.hovered.contains(&id) {
+                    return true;
+                }
             }
         }
         false
     }
 }
 
+pub enum InteractionState {
+    Default,
+    Hover,
+    // TODO: focus, active, disabled
```

Comment:
What if the ```View``` is both focused and hovered?

Reply:
I went with this approach because I think its simpler and cleaner for computing the styles - focus simply will have a higher priority than hover. 
I think if the goal is to support combining interaction states for the styles(e.g. style for hover+focus) then the flag approach is the way  - but it also adds more complexity. For example `is_disabled` needs to be checked before applying other flags, because it doesn't make sense to have a disabled view that is active/focused etc.

Reply:
I think the flags are useful for other stuff, e.g., ```is_disabled``` can be used by ```event```. 

Also, focus and active are already taken care of in ```AppState```. 

Back to the style, I'm not sure if we should combine all of the states or only the higher priority state. Do you know how web does that?

Reply:
In web the styles are combined and the higher priority selectors override the lower priority, but you can also combine selectors. 
Suppose an input is hovered **and** focused -  `:hover:focus` applies its styles on top of  `:focus`  which overrides `:hover` which overrides the base styles.

My concern is if we do it this way, how will the user API work for combining the states(e.g. hover + focus) when styling views? In web its quite easy to combine them (since they are just strings - "input:hover:focus" etc.). Maybe the interactive state can be exposed, so it would be possible to do like so :
 ```rust
 stack()
style(......),
hover_style(.....),
focus_style(......)
chained_style(cx, [
    (InteractionState.Hovered, InteractionState.Focused, || Style::default().background(Color::YELLOW)), 
    (InteractionState.Disabled, InteractionState.Hovered, || Style::default().background(Color::BLACK))
   // the latter one overrides previous, if both match 
]);
 ```

Reply:
It's quite complex :) I guess we'll worry about the combined selectors later.

If you don't mind changing the hover to be a flag, and we get this RP merged first.